cargo-audit: Build fails with `--no-default-features`

At the tag `cargo-audit/v0.21.0-pre.0` and at current `main`, commit 0480bccdc3109ab0131f07f49c131071f1ec903e, running, in `cargo-audit`:

```sh
cargo build --no-default-features
```

Fails with:

```
error[E0433]: failed to resolve: use of undeclared crate or module `cargo_lock`
  --> cargo-audit/src/error.rs:98:11
   |
98 | impl From<cargo_lock::Error> for Error {
   |           ^^^^^^^^^^ use of undeclared crate or module `cargo_lock`
   |
help: consider importing this crate
   |
3  + use rustsec::cargo_lock;
   |
```

...and other related errors.

cc @Shnatsel

--- a/.github/workflows/cargo-audit.yml
+++ b/.github/workflows/cargo-audit.yml
@@ -50,6 +50,8 @@ jobs:
           profile: minimal
           override: true
       - uses: Swatinem/rust-cache@v2
+      - run: cargo test --no-default-features
+      - run: cargo test
       - run: cargo test --all-features
 
   doc:

--- a/cargo-audit/Cargo.toml
+++ b/cargo-audit/Cargo.toml
@@ -18,6 +18,7 @@ maintenance = { status = "actively-developed" }
 
 [dependencies]
 abscissa_core = { workspace = true }
+cargo-lock = { workspace = true }
 clap = { workspace = true }
 home = { workspace = true }
 rustsec = { workspace = true, features = ["dependency-tree"] }
@@ -27,7 +28,6 @@ thiserror = { workspace = true }
 
 # for scanning binary files
 auditable-info = { workspace = true, features = ["wasm"], optional = true }
-cargo-lock = { workspace = true, optional = true }
 auditable-serde = { workspace = true, optional = true }
 quitters = { workspace = true, optional = true }
 once_cell = { workspace = true, optional = true }
@@ -43,4 +43,4 @@ toml = { workspace = true }
 [features]
 default = ["binary-scanning"]
 fix = []
-binary-scanning = ["dep:auditable-info", "dep:cargo-lock", "dep:auditable-serde", "dep:binfarce", "dep:quitters", "dep:once_cell"]
+binary-scanning = ["dep:auditable-info", "dep:auditable-serde", "dep:binfarce", "dep:quitters", "dep:once_cell"]

--- a/cargo-audit/src/binary_format.rs
+++ b/cargo-audit/src/binary_format.rs
@@ -1,5 +1,6 @@
 /// A shim around `binfarce::Format` so that `binfarce` crate could be an optional dependency
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[allow(dead_code)] // May be unused when the "binary-scanning" feature is disabled
 pub enum BinaryFormat {
     Elf32,
     Elf64,

